app/src/main/java/fr/free/nrw/commons/upload/UploadModel.java

```diff
@@ -205,6 +205,7 @@ public void updateUploadItem(int index, UploadItem uploadItem) {
         UploadItem uploadItem1 = items.get(index);
         uploadItem1.setDescriptions(uploadItem.descriptions);
         uploadItem1.setTitle(uploadItem.title);
+        store.putString("Title", uploadItem.place.getName());
```

Comment:
It would be better if you could pass along this data, setting/retrieving a value in sharedprefences is a bit surprising

Reply:
@macgills Can you please elaborate:-)
showSuccessToast method in WikiEditService class uses sharedpreference to retrieve the value for title. If not set it uses default, that's why i set the value when item is uploaded.

Reply:
Yeah, whoever did that just shouldn't have. We shouldn't use shared preferences to communicate data unless it is actually a preference of the app/user.

Add `wikiItemName` as a parameter to `showSuccessToast()` and keep adding it up the chain of methods until you have access to it.

I still have no idea what half these fields are or aren't so I can't comment if `place.name()` is a wikiItemName